Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-100.ca-central-1.compute.internal
Sha: a44c3ccb2cfefe966380731c74b6fb4f264da67b
Number of Error(s): 67
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1441:6: error: expected \xe2\x80\x98;\xe2\x80\x99 after class definition
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1402:17: error: \xe2\x80\x98doCloseExpired\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1443:5: error: \xe2\x80\x98virtual\xe2\x80\x99 outside class declaration
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1445:9: error: \xe2\x80\x98closing\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1446:13: error: \xe2\x80\x98started\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1448:13: error: \xe2\x80\x98toCopy\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1449:13: error: \xe2\x80\x98toClose\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1450:13: error: \xe2\x80\x98bct\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1451:13: error: \xe2\x80\x98hct\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1462:5: error: \xe2\x80\x98virtual\xe2\x80\x99 outside class declaration
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1464:16: error: \xe2\x80\x98aborting\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1467:5: error: \xe2\x80\x98virtual\xe2\x80\x99 outside class declaration
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1467:49: error: non-member function \xe2\x80\x98void removeCache(ILazyFileIO*)\xe2\x80\x99 cannot have cv-qualifier
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1469:25: error: \xe2\x80\x98crit\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1473:29: error: \xe2\x80\x98files\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1476:31: error: \xe2\x80\x98todo\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1486:5: error: \xe2\x80\x98virtual\xe2\x80\x99 outside class declaration
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1521:29: error: \xe2\x80\x98crit\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1522:35: error: \xe2\x80\x98files\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1541:29: error: \xe2\x80\x98fileErrorList\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1561:24: error: \xe2\x80\x98openFile\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/ccdfile.cpp:1572:29: error: \xe2\x80\x98doCopy\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14295/HPCC-Platform/roxie/ccd/
 ... ( comment is too long, 90 lines, 10899 bytes, truncated) ... 
[Build error:](https://gist.github.com/fda5bf692d718580796feec42b3c22fa)
Build log [Build log](https://gist.github.com/HPCCSmoketest/fda5bf692d718580796feec42b3c22fa#file-result-20-11-12-13-42-20-log)

```

--- a/roxie/ccd/ccdfile.cpp
+++ b/roxie/ccd/ccdfile.cpp
@@ -783,6 +783,7 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
     bool closePending[2];
     StringAttrMapping fileErrorList;
 #ifdef _CONTAINERIZED
+    bool cidtActive = false;
     Semaphore cidtStarted;
 #endif
     Semaphore bctStarted;
@@ -1257,6 +1258,7 @@ public:
         {
             cidt.start();
             cidtStarted.wait();
+            cidtActive = true;
         }
 #endif
     }
@@ -1425,14 +1427,14 @@ public:
             toClose.interrupt();
             bct.join(timeout);
             hct.join(timeout);
-#ifdef _CONTAINERIZED
-            if (activeCacheReportingBuffer && cacheReportPeriodSeconds)
-            {
-                cidtSleep.interrupt();
-                cidt.join(timeout);
-            }
-#endif
         }
+#ifdef _CONTAINERIZED
+        if (cidtActive && activeCacheReportingBuffer && cacheReportPeriodSeconds)
+        {
+            cidtSleep.interrupt();
+            cidt.join(timeout);
+        }
+#endif
     }
 
     virtual void wait()
@@ -1444,14 +1446,14 @@ public:
             toClose.signal();
             bct.join();
             hct.join();
-#ifdef _CONTAINERIZED
-            if (activeCacheReportingBuffer && cacheReportPeriodSeconds)
-            {
-                cidtSleep.signal();
-                cidt.join();
-            }
-#endif
         }
+#ifdef _CONTAINERIZED
+        if (cidtActive && activeCacheReportingBuffer && cacheReportPeriodSeconds)
+        {
+            cidtSleep.signal();
+            cidt.join();
+        }
+#endif
     }
 
     virtual CFPmode onProgress(unsigned __int64 sizeDone, unsigned __int64 totalSize)

--- a/roxie/ccd/ccdmain.cpp
+++ b/roxie/ccd/ccdmain.cpp
@@ -917,7 +917,6 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         flushJHtreeCacheOnOOM = topology->getPropBool("@flushJHtreeCacheOnOOM", true);
         fastLaneQueue = topology->getPropBool("@fastLaneQueue", true);
         udpOutQsPriority = topology->getPropInt("@udpOutQsPriority", 0);
-        udpSnifferEnabled = topology->getPropBool("@udpSnifferEnabled", true);
         udpRetryBusySenders = topology->getPropInt("@udpRetryBusySenders", 0);
 
         // Historically, this was specified in seconds. Assume any value <= 10 is a legacy value specified in seconds!
@@ -946,13 +945,13 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         udpLocalWriteSocketSize = topology->getPropInt("@udpLocalWriteSocketSize", 1024000);
 #ifndef _CONTAINERIZED
         roxieMulticastEnabled = topology->getPropBool("@roxieMulticastEnabled", true) && !useAeron;   // enable use of multicast for sending requests to agents
-#endif
+        udpSnifferEnabled = topology->getPropBool("@udpSnifferEnabled", roxieMulticastEnabled);
         if (udpSnifferEnabled && !roxieMulticastEnabled)
         {
             DBGLOG("WARNING: ignoring udpSnifferEnabled setting as multicast not enabled");
             udpSnifferEnabled = false;
         }
-
+#endif
         int ttlTmp = topology->getPropInt("@multicastTTL", 1);
         if (ttlTmp < 0)
         {
@@ -1123,7 +1122,6 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
                 queryDirectory.append(codeDirectory).append("queries");
         }
         addNonEmptyPathSepChar(queryDirectory);
-        queryFileCache().start();
         getTempFilePath(tempDirectory, "roxie", topology);
 
 #ifdef _WIN32
@@ -1188,7 +1186,10 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         enableForceRemoteReads(); // forces file reads to be remote reads if they match environment setting 'forceRemotePattern' pattern.
 
         if (!oneShotRoxie)
+        {
+            queryFileCache().start();
             loadPlugins();
+        }
         unsigned snifferChannel = numChannels+2; // MORE - why +2 not +1??
 #ifdef _CONTAINERIZED
         initializeTopology(topoValues, myRoles);

--- a/roxie/ccd/ccdqueue.cpp
+++ b/roxie/ccd/ccdqueue.cpp
@@ -2615,49 +2615,58 @@ public:
         Owned<StringContextLogger> logctx = new StringContextLogger("PacketDiscarder");
         rowManager.setown(roxiemem::createRowManager(0, NULL, *logctx, NULL, false));
         mc.setown(ROQ->queryReceiveManager()->createMessageCollator(rowManager, RUID_DISCARD));
-        while (!aborted)
+        try
         {
-            bool anyActivity = false;
-            Owned<IMessageResult> mr = mc->getNextResult(5000, anyActivity);
-            if (mr)
+            while (!aborted)
             {
-                if (traceLevel > 4)
-                    DBGLOG("Discarding unwanted message");
-                unsigned headerLen;
-                const RoxiePacketHeader &header = *(const RoxiePacketHeader *) mr->getMessageHeader(headerLen);
-                if (headerLen)
+                bool anyActivity = false;
+                Owned<IMessageResult> mr = mc->getNextResult(5000, anyActivity);
+                if (mr)
                 {
-                    switch (header.activityId)
+                    if (traceLevel > 4)
+                        DBGLOG("Discarding unwanted message");
+                    unsigned headerLen;
+                    const RoxiePacketHeader &header = *(const RoxiePacketHeader *) mr->getMessageHeader(headerLen);
+                    if (headerLen)
                     {
-                        case ROXIE_FILECALLBACK:
+                        switch (header.activityId)
                         {
-                            Owned<IMessageUnpackCursor> callbackData = mr->getCursor(rowManager);
-                            OwnedConstRoxieRow len = callbackData->getNext(sizeof(RecordLengthType));
-                            if (len)
+                            case ROXIE_FILECALLBACK:
                             {
-                                RecordLengthType *rowlen = (RecordLengthType *) len.get();
-                                OwnedConstRoxieRow row = callbackData->getNext(*rowlen);
-                                const char *rowdata = (const char *) row.get();
-                                // bool isOpt = * (bool *) rowdata;
-                                // bool isLocal = * (bool *) (rowdata+1);
-                                ROQ->sendAbortCallback(header, rowdata+2, *logctx);
+                                Owned<IMessageUnpackCursor> callbackData = mr->getCursor(rowManager);
+                                OwnedConstRoxieRow len = callbackData->getNext(sizeof(RecordLengthType));
+                                if (len)
+                                {
+                                    RecordLengthType *rowlen = (RecordLengthType *) len.get();
+                                    OwnedConstRoxieRow row = callbackData->getNext(*rowlen);
+                                    const char *rowdata = (const char *) row.get();
+                                    // bool isOpt = * (bool *) rowdata;
+                                    // bool isLocal = * (bool *) (rowdata+1);
+                                    ROQ->sendAbortCallback(header, rowdata+2, *logctx);
+                                }
+                                else
+                                    DBGLOG("Unrecognized format in discarded file callback");
+                                break;
                             }
-                            else
-                                DBGLOG("Unrecognized format in discarded file callback");
-                            break;
+                            // MORE - ROXIE_ALIVE perhaps should go here too? debug callbacks? Actually any standard query results should too (though by the time I see them here it's too late (that may change once start streaming)
                         }
-                        // MORE - ROXIE_ALIVE perhaps should go here too? debug callbacks? Actually any standard query results should too (though by the time I see them here it's too late (that may change once start streaming)
                     }
+                    else
+                        DBGLOG("Unwanted message had no header?!");
                 }
-                else
-                    DBGLOG("Unwanted message had no header?!");
+                else if (!anyActivity)
+                {
+                    // to avoid leaking partial unwanted packets, we clear out mc periodically...
+                    ROQ->queryReceiveManager()->detachCollator(mc);
+                    mc.setown(ROQ->queryReceiveManager()->createMessageCollator(rowManager, RUID_DISCARD));
+                }
             }
-            else if (!anyActivity)
-            {
-                // to avoid leaking partial unwanted packets, we clear out mc periodically...
-                ROQ->queryReceiveManager()->detachCollator(mc);
-                mc.setown(ROQ->queryReceiveManager()->createMessageCollator(rowManager, RUID_DISCARD));
-            }
+        }
+        catch (IException * E)
+        {
+            if (!aborted || QUERYINTERFACE(E, InterruptedSemaphoreException) == NULL)
+                EXCLOG(E);
+            ::Release(E);
         }
         return 0;
     }

--- a/roxie/ccd/ccdstate.cpp
+++ b/roxie/ccd/ccdstate.cpp
@@ -1854,8 +1854,6 @@ private:
 
 class CRoxiePackageSetManager : implements IRoxieQueryPackageManagerSet, implements ISafeSDSSubscription, public CInterface
 {
-    Owned<IDaliPackageWatcher> pSetsNotifier;
-    Owned<IDaliPackageWatcher> pMapsNotifier;
 public:
     IMPLEMENT_IINTERFACE;
     CRoxiePackageSetManager(const IQueryDll *_standAloneDll) :
@@ -1876,6 +1874,10 @@ public:
     {
         autoReloadThread.stop();
         autoReloadThread.join();
+        if (pSetsNotifier)
+            daliHelper->releaseSubscription(pSetsNotifier);
+        if (pMapsNotifier)
+            daliHelper->releaseSubscription(pMapsNotifier);
     }
 
     virtual ISafeSDSSubscription *linkIfAlive() override { return isAliveAndLink() ? this : nullptr; }
@@ -1959,6 +1961,8 @@ private:
     Owned<const IQueryDll> standAloneDll;
     Owned<CRoxieDebugSessionManager> debugSessionManager;
     Owned<IRoxieDaliHelper> daliHelper;
+    Owned<IDaliPackageWatcher> pSetsNotifier;
+    Owned<IDaliPackageWatcher> pMapsNotifier;
     mutable ReadWriteLock packageCrit;
     InterruptableSemaphore controlSem;
     Owned<CRoxiePackageSetWatcher> allQueryPackages;

--- a/roxie/udplib/udptrs.cpp
+++ b/roxie/udplib/udptrs.cpp
@@ -37,7 +37,7 @@ unsigned udpOutQsPriority = 0;
 unsigned udpMaxRetryTimedoutReqs = 0; // 0 means off (keep retrying forever)
 unsigned udpRequestToSendTimeout = 0; // value in milliseconds - 0 means calculate from query timeouts
 unsigned udpRequestToSendAckTimeout = 10; // value in milliseconds
-bool udpSnifferEnabled = true;
+bool udpSnifferEnabled = false;
 
 using roxiemem::DataBuffer;
 // MORE - why use DataBuffers on output side?? We could use zeroCopy techniques if we had a dedicated memory area.

--- a/system/jlib/jdebug.cpp
+++ b/system/jlib/jdebug.cpp
@@ -1859,7 +1859,7 @@ void OsDiskInfo::initMajorMinor()
     // and match those with entries in /proc/diskstats
     StringBuffer cmd("lsblk -o TYPE,MAJ:MIN --pairs");
     Owned<IPipeProcess> pipe = createPipeProcess();
-    if (pipe->run("list disks", cmd, nullptr, false, true, true, 8192))
+    if (pipe->run(nullptr, cmd, nullptr, false, true, true, 8192))
     {
         StringBuffer output;
         Owned<ISimpleReadStream> pipeReader = pipe->getOutputStream();

